src/ruby/puppetserver-lib/puppet/server/logger.rb

```diff
@@ -27,5 +27,25 @@ def handle(msg)
 class Puppet::Server::Logger
   def self.init_logging
     Puppet::Util::Log.newdestination(:logback)
+    Puppet[:log_level] = level_from_logback(get_logger)
+  end
+
+  def self.level_from_logback(logger)
+    case
+    when logger.isDebugEnabled()
+      'debug'
```

Comment:
should these be symbols?

Reply:
It appears to have previously been a string (see the last file in this diff).

Reply:
indeed

Reply:
Puppetserver expects that this option will be set on the commandline and thus will be a string; if you pass a symbol Puppet explodes.